logdna/resource_archive.go

```diff
@@ -0,0 +1,278 @@
+package logdna
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const archiveConfigID = "archive"
+
+type archiveConfig struct {
+	Integration        string `json:"integration,omitempty"`
+	Bucket             string `json:"bucket,omitempty"`
+	Endpoint           string `json:"endpoint,omitempty"`
+	Apikey             string `json:"apikey,omitempty"`
+	Resourceinstanceid string `json:"resourceinstanceid,omitempty"`
+	Accountname        string `json:"accountname,omitempty"`
+	Accountkey         string `json:"accountkey,omitempty"`
+	Projectid          string `json:"projectid,omitempty"`
+	Space              string `json:"space,omitempty"`
+	Accesskey          string `json:"accesskey,omitempty"`
+	Secretkey          string `json:"secretkey,omitempty"`
+	Authurl            string `json:"authurl,omitempty"`
+	Expires            int    `json:"expires,omitempty"`
+	Username           string `json:"username,omitempty"`
+	Password           string `json:"password,omitempty"`
+	Tenantname         string `json:"tenantname,omitempty"`
```

Comment:
nit: conventionally, struct fields should be named:

* mixed case, e.g. `AccountName`, `AccountKey`
* initialisms should be all caps, e.g. `APIKey`, `ProjectID`

Reply:
I did not inject additional context/meaning to the fields by distinguishing words and initials because neither the API fields nor the provider fields do so. (This is also 'under the hood') However I can definitely add context and casing to the fields if you believe it's needed.

Reply:
changed!